src/views/Villager/Villager.test.jsx

```diff
@@ -0,0 +1,15 @@
+import { render, screen } from '@testing-library/react';
+import { MemoryRouter, Route } from 'react-router-dom';
+import Villager from './Villager';
+
+test('detail page loads the correct villager', async () => {
+  render(
+    <MemoryRouter initialEntries={[{ pathname: '/4' }]}>
+      <Route path="/:id">
```

Comment:
It's not ideal to have a detail id right outside the home path. Ideally it would be something like /villagers/:id. 
This comment would be better placed in your App.js where your route is actually defined. 😄